vision_datasets/kv_pair/manifest.py

```diff
@@ -0,0 +1,127 @@
+from typing import Optional
+from ..common import ImageLabelManifest, AnnotationWiseDatasetManifest, DatasetTypes
+        
+
+class FieldSchema:
+    TYPE_NAME_TO_PYTHON_TYPE = {
+        'array': list,
+        'object': dict,
+        'number': (float, int),
+        'integer': int,
+        'boolean': bool,
+        'string': str, 
+        'boundingBox': list
+    }
+    
+    def __init__(self, type: str,
+                 description: str = None,
+                 examples: list[str] = None,
+                 enum: list[str] = None,
+                 items: 'FieldSchema' = None,
+                 properties: dict[str, 'FieldSchema'] = None) -> None:
+        
+        self.type = type
+        self.description = description
+        self.examples = examples
+        self.enum = enum
+        self.items = FieldSchema(**items) if items else None
+        self.properties = {k: FieldSchema(**v) for k, v in properties.items()} if properties else None
+        self._check()
+    
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, FieldSchema):
+            return False
+        return (self.type == other.type \
+            and self.description == other.description \
+            and self.examples == other.examples \
+            and self.enum == other.enum \
+            and self.items == other.items \
+            and self.properties == other.properties)
+    
+    def _check(self):
+        if self.type not in self.TYPE_NAME_TO_PYTHON_TYPE:
+            raise ValueError(f'Invalid type: {self.type}')
+        if self.enum and self.type not in ['string', 'number', 'integer']:
+            raise ValueError(f'enum is only allowed for string, number, integer types')
+        if self.type == 'array' and not self.items:
+            raise ValueError(f'items must be provided for array type')
+        elif self.type == 'object' and not self.properties:
+            raise ValueError(f'properties must be provided for object type')
+
+
+class Schema:
```

Comment:
To make this clear in other parts of the code, could we name this `KVPairSchema`?